docs/abcd_rooparametrichist_tutorial/rooparametrichist_exercise.md

```diff
@@ -0,0 +1,519 @@
+getattr(ws, "import")(param_Bkg_D_norm, RooFit.Rename("bkg_D"+"_norm"),RooFit.RecycleConflictNodes())
+
+```
+</details>
+
+For the signal region A, create a ```RooParametricHist``` with each bin made from a ```RooFormulaVar``` relating the C,D,B regions to A via the ABCD formula. 
+
+<details>
+<summary> Create RooParametricHist for SR background template  </summary>
+
+```
+#Relate the signal region (A) to control region B via transfer factors
+#Define RooArgList for expected yields in region A after applying transfer factors
+process_AB_region_bins = RooArgList()
+#Define list for transfer factors
+TF_list = []
+#Define list for expected yields in region A after applying transfer factors
+process_AB_region_bins_list = []
+
+#Compute per-bin transfer factor
+#Loop over the bins of the transfering region B, and compute the transfer factors as C/D
+for i in range(1,histB_pr.GetNbinsX()+1):
+    #Define transfer factor as a RooFormulaVar. Use the method .obj for RooWorkSpace to retrieve the yield for a given bin and region
+    TF_i = RooFormulaVar("TF"+str(i),"Transfer factor C/D bin " + str(i),"(@0/@1)",RooArgList(ws.obj("Bkg_C_region_bin_"+str(i)) , ws.obj("Bkg_D_region_bin_"+str(i)) ))
+    TF_list.append(TF_i)
+    #Compute the expected yield in the signal region as A = B * TF. This will be used to initialise the RooParametricHist in the Signal Region
+    bin_AB_i = RooFormulaVar("Bkg_AB_region_bin_"+str(i),"Background yield in SR A region bin " + str(i), "@0*@1", RooArgList(TF_i, ws.obj("Bkg_B_region_bin_"+str(i)) ))
+    process_AB_region_bins_list.append(bin_AB_i)
+    process_AB_region_bins.add(process_AB_region_bins_list[i-1])
+
+#Create parametric histogram for signal region (A) using bin contents saved in process_AB_region_bins
+param_hist_A_region = RooParametricHist("bkg_A", "Background PDF in A region",variable_z,process_AB_region_bins,histB_pr)
+
+#Define total normalization parameter
+param_bkg_A_norm = RooAddition("bkg_A"+"_norm","Total Number of events from background in A region",process_AB_region_bins)
+
+#Store in Workspace
+getattr(ws, "import")(param_hist_A_region, RooFit.Rename("bkg_A"))
+getattr(ws, "import")(param_bkg_A_norm, RooFit.Rename("bkg_A"+"_norm"),RooFit.RecycleConflictNodes())
+
+```
+</details>
+
+To run the workspace creation script:
+
+```
+python3 utils/create_workspace.py -m 1500
+
+```
+
+where ```-m``` is the flag for the mass point you want to run the script on. The script will use by default the histograms stored in ```generated_histograms``` in the tutorial directory.
+After running the script, the workspace will be saved in ```example_analysis/datacards/```. To create the datacards automatically fatching the correct workspace, run:
+
+
+```
+python3 utils/create_datacards.py -m 1500
+
+```
+
+The datacards will be created in the directory ```example_analysis/datacards/mPhi1500/``` inside the tutorial directory. The datacards can be combined then using the usual command:
```

Comment:
Can we just make sure to add a statement to go change into this directory (or add the cd command below, or both).

Reply:
i have updated the text as you suggested by adding explicit instruction to enter the directory